### ISSUE TITLE
clickhouse-client can't use --password option

### ISSUE THREAD

**Issue by Livenux** [yandex/ClickHouse#3574]
``` bash
clickhouse-client --password
```
> Bad arguments: the required argument for option '--password' is missing

I set the username and password for clickhosue, but the client cannot use the --password option. Is this a client error?

**Comment by blinkov** [comment]
This is indeed looks like incorrect command to run. You should either specify password value like `--password=12345` (which is often not a good idea since it'll be visible in tools like `ps` or `top`), or use `--ask-password` if you want to enter it interactively.

**Comment by alexey-milovidov** [comment]
We can implement it for compatibility with mysql CLI:

```
sudo mysql -uroot --host=127.0.0.1 --password
Enter password:
```

**Comment by alexey-milovidov** [comment]
Implemented in #4230
